examples/hitl/rearrange_v2/world.py

```diff
@@ -0,0 +1,122 @@
+
+
+class World:
+    """
+    Global world state shared by each user.
+    """
+
+    def __init__(
+        self,
+        sim: RearrangeSim,
+    ):
+        self._sim = sim
+
+        # Cache of all held objects.
+        # Prevents users from picking objects held by others.
+        self._all_held_object_ids: Set[int] = set()
+        # Cache of all opened articulated object links.
+        self._opened_link_set: Set = set()
+        # Cache of all link IDs and their parent articulated objects.
+        # Used to speed-up sim queries.
+        self._link_id_to_ao_map: Dict[int, int] = {}
+        # Cache of pickable objects IDs.
+        self._pickable_object_ids: Set[int] = set()
+        # Cache of interactable objects IDs.
+        self._interactable_object_ids: Set[int] = set()
+        # Cache of agent articulated object IDs.
+        self._agent_object_ids: Set[int] = set()
+
+    def reset(self) -> None:
+        """
+        Reset the world state. Call every time the scene changes.
+        """
+        sim = self._sim
+        self._all_held_object_ids = set()
+        self._opened_link_set = set()
+        self._link_id_to_ao_map = sim_utilities.get_ao_link_id_map(sim)
+
+        # Find pickable objects.
+        self._pickable_object_ids = set(sim._scene_obj_ids)
+        for pickable_obj_id in self._pickable_object_ids:
+            rigid_obj = self.get_rigid_object(pickable_obj_id)
+            # Ensure that rigid objects are collidable.
+            rigid_obj.collidable = True
+
+        # Get set of interactable articulated object links.
+        # Exclude all agents.
+        agent_articulated_objects = set()
+        agent_manager: ArticulatedAgentManager = sim.agents_mgr
+        for agent_index in range(len(agent_manager)):
+            agent = agent_manager[agent_index]
+            agent_ao = agent.articulated_agent.sim_obj
+            agent_articulated_objects.add(agent_ao.object_id)
+            self._agent_object_ids.add(agent_ao.object_id)
+        self._interactable_object_ids = set()
+        aom = sim.get_articulated_object_manager()
+        all_ao: List[
+            ManagedArticulatedObject
+        ] = aom.get_objects_by_handle_substring().values()
+        # Classify all non-root links.
+        for ao in all_ao:
+            for link_object_id in ao.link_object_ids:
+                if link_object_id != ao.object_id:
+                    # Link is part of an agent.
+                    if ao.object_id in agent_articulated_objects:
+                        self._agent_object_ids.add(link_object_id)
+                    # Link is not part of an agent.
+                    else:
+                        self._interactable_object_ids.add(link_object_id)
+                        # Make sure that link is in "closed" state.
+                        link_index = self.get_link_index(link_object_id)
+                        if link_index:
+                            sim_utilities.close_link(ao, link_index)
```

Comment:
Is this necessary? Are scenes being initialized with open links?
It could be that non-zero joint states come from a scene instance on purpose, so this would change the expected scene state. Worth some thought.